toolkit/indoors/src/main/java/com/arcgismaps/toolkit/indoors/FloorFilterState.kt

```diff
@@ -212,30 +226,59 @@ private class FloorFilterStateImpl(
             }
         }
 
-    init {
-        coroutineScope.launch {
-            loadFloorManager()
-        }
-    }
-
     /**
-     * Loads the floorManager when the Map is loaded.
+     * Initializes the state object by loading the GeoModel and the GeoModel's FloorManager.
      *
-     * @since 200.2.0
+     * @return the [Result] indicating if the initialization was successful or not
+     * @since 200.6.0
      */
-    private suspend fun loadFloorManager() {
+    override suspend fun initialize(): Result<Unit> = runCatchingCancellable {
+        if (_initializationStatus.value is InitializationStatus.Initialized) {
+            return Result.success(Unit)
+        }
+        _initializationStatus.value = InitializationStatus.Initializing
         geoModel.load().onSuccess {
-            val floorManager: FloorManager = geoModel.floorManager
-                ?: return
+            val floorManager: FloorManager? = geoModel.floorManager
+            if (floorManager == null) {
+                val error = FloorFilterException(FloorFilterError.GEOMODEL_HAS_NO_FLOOR_AWARE_DATA)
+                _initializationStatus.value = InitializationStatus.FailedToInitialize(error)
+                throw error
+            }
             floorManager.load().onSuccess {
                 _floorManager.value = floorManager
+                // make sure the UI gets updated with the values for SelectedSiteId, SelectedFacilityId
+                // and SelectedLevelId that were applied before initialization
+                setSelectedValuesAppliedBeforeInitialization()
                 // no FloorLevel is selected at this point, so clear the FloorFilter from the selected GeoModel
                 filterMap()
+                _initializationStatus.value = InitializationStatus.Initialized
+                return Result.success(Unit)
             }.onFailure {
-                return
+                _initializationStatus.value = InitializationStatus.FailedToInitialize(it)
+                throw it
             }
         }.onFailure {
-            return
+            _initializationStatus.value = InitializationStatus.FailedToInitialize(it)
+            throw it
+        }
+    }
+
+    /**
+     * Sets the selected values that were applied before initialization by the user.
+     *
+     * @since 200.6.0
+     */
+    private fun setSelectedValuesAppliedBeforeInitialization() {
+        if (_selectedSiteId != null) {
+            selectedSiteId = _selectedSiteId
+        }
+        if (_selectedFacilityId != null) {
+            selectedFacilityId = _selectedFacilityId
+        }
+        if (_selectedLevelId != null) {
+            val temp = _selectedLevelId
+            _selectedLevelId = null
+            selectedLevelId = temp
```

Comment:
I'm not sure if I understand why the internal `_selectedLevelId` is being diverted to `null` here. 
```suggestion
            selectedLevelId = _selectedLevelId
```

Reply:
We have a check were we don't set the selectedLevelId to a new value if it is the same as the old one. This logic caches the old value, sets the backing property to null and resets the selectedLevelId for setting the same value again